src/main/g8/src/main/scala/com/example/WebServer.scala

```diff
@@ -0,0 +1,34 @@
+package com.example
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.server.Directives._
+import akka.stream.ActorMaterializer
+import scala.io.StdIn
+
+object WebServer {
+  def main(args: Array[String]) {
+
+    implicit val system = ActorSystem("my-system")
+    implicit val materializer = ActorMaterializer()
+    // needed for the future flatMap/onComplete in the end
+    implicit val executionContext = system.dispatcher
+
+    val bindingFuture = Http().bindAndHandle(route, "localhost", 8080)
+
+    println(s"Server online at http://localhost:8080/\nPress RETURN to stop...")
+    StdIn.readLine() // let it run until user presses return
```

Comment:
I'm not sure I understand the purpose of this.
This readline main purpose is to signal server shutdown. I guess I could put the readline together with http unbinding and actor system termination.